### PR TITLE
Adding workoad-identity support for Azure

### PR DESCRIPTION
**Pull request by tkennes**
## Description of Change
Adding Azure Workload Identity option

## Checklist

- [x] PR description included and stakeholders cc'd
- [ ] tests are changed or added
- [x] yarn test passes
- [x] yarn lint has been run
- [ ] git pre-commit hook is successfully executed.
      (**Please refrain from using `--no-verify`**)
- [ ] yarn changeset was run and relevant packages have been updated as a major/minor/patch bump with descriptions
      (**Determine version update using [Semantic Versioning](https://semver.org/)**)
- [x] relevant documentation is changed or added

## Notes

Additional information relevant to the changes

### CODE REVIEW

File: packages/azure/src/application/AzureCredentialsProvider.ts [hunk 1]
```diff
@@ -23,6 +26,11 @@ export default class AzureCredentialsProvider {
           clientIdFromGoogle,
           clientSecretFromGoogle,
         )
+      case 'workload_identity':
```
Comment (4upz) [inline packages/azure/src/application/AzureCredentialsProvider.ts]: To keep this consistent with config naming conventions, this should probably be capitalized as `WORKLOAD_IDENTITY`.

**Review by 4upz — CHANGES_REQUESTED** [review]
@tkennes Thank you for this contribution! I only had one suggestion on the implementation, otherwise it looks great!

Before merging, would it be possible to add a test case for this in `packages/azure/src/__tests__/AzureCredentialsProvider.test.ts`?

**Comment by tkennes** [comment]
@4upz , I'll take it, good suggestion! I'm still a bit struggling getting the tests positive, don't really understand why all these tests need to run for this kind of change, but perhaps I'll push it without testing and ask for help. :)

As for the test case, I'm not sure whether it would make sense to create a separate test for this, since it relies on the Azure SDK for authentication, similar to the default one. Do you agree?

**Comment by tkennes** [comment]
@4upz , I just pushed while skipping the precommit tests

File: docker-compose.azure.yml
Comment (4upz) [inline docker-compose.azure.yml:1]: Is this a necessary addition for this feature? It seems like it is the normal docker-compose with some configs/secrets removed

**Review by 4upz — CHANGES_REQUESTED** [review]

**Comment by 4upz** [comment]
> @4upz , I just pushed while skipping the precommit tests

Thanks @tkennes! I only added one remaining question. No worries on the test case. You are right, it would be a test case added just to check that the condition calls the mocked Azure library. As long as the coverage threshold is met, we can skip it

File: docker-compose.azure.yml
Comment (tkennes) [reply docker-compose.azure.yml:1]: Yeah, fair enough, I used it for some testing!

**Review by 4upz — APPROVED** [review]
Looks great! Thanks for making the changes! I'll go ahead and merge this in

**Comment by tkennes** [comment]
@4upz, how about documentation? Perhaps we can add some notes there as well?

**Comment by 4upz** [comment]
> @4upz, how about documentation? Perhaps we can add some notes there as well?

Yes, it will definitely need documentation! It can be added as an additional option within the [Azure Authentication docs](https://www.cloudcarbonfootprint.org/docs/azure#options-for-azure-authentication). That can be added to the `microsite/docs/ConnectingData/Azure.md`. Would you be interested in taking a stab at it? It can be submitted as a separate PR.

**Comment by tkennes** [comment]
> > @4upz, how about documentation? Perhaps we can add some notes there as well?
> 
> Yes, it will definitely need documentation! It can be added as an additional option within the [Azure Authentication docs](https://www.cloudcarbonfootprint.org/docs/azure#options-for-azure-authentication). That can be added to the `microsite/docs/ConnectingData/Azure.md`. Would you be interested in taking a stab at it?

Sure! 

I'm also curious to meet some of the people behind the product. I have a couple of other ideas on my mind that could make adoption a lot easier, especially when working with Azure. Is there some sort of recurrent project-meeting I could join? :)

**Comment by 4upz** [comment]
@tkennes Awesome, thanks! Feel free to let me know when it's ready for review.

As for the meeting, we'd be happy to! Are you in any NA timezones? If so, we have a recurring Dev meeting on Tuesdays that I can invite you to. Otherwise, happy to chat via email or LinkedIn and organize a separate time

**Comment by tkennes** [comment]
@4upz, sure, will do!

Great, happy to hear that! No, unfortunately not, I'm based in Amsterdam, so CEST+1. What time are the Dev meetings? Feel free to reach out via email (t.a.j.m.kennes@gmail.com, or linkedin (Tom Kennes)).

**Pull request merged**